algotaurus.py

```diff
@@ -529,12 +560,13 @@ def __init__(self, size=15, lines=30):
         self.timerlabel = ttk.Label(self.speedframe, textvariable=self.rt_str)
         self.textPad.bind('<Button-3>', self.rclick)
         self.textPad.bind('<Key>', self.validate_input)        
-        # Creating canvas and drawing sample labyrinth
-        self.canvas = tk.Canvas(self.mainframe, width=self.size*(self.x+4), height=self.size*(self.y+4))
-        samplab = Labyrinth(self.x, self.y)
-        sample = samplab.labyr
-        Robot(samplab)
-        self.draw_labyr(sample)
+        # Creating canvas and drawing the labyrinth
+        self.canvas = tk.Canvas(self.mainframe, width=self.size*(self.maze_columns + 4), height=self.size * (self.maze_rows + 4))        
+        self.labyr = Labyrinth(self.maze_columns, self.maze_rows)
+        self.robot = Robot(self.labyr)
+        self.robot_face = self.labyr.labyr.max()
+        self.robot_pos = np.where(self.labyr.labyr == self.robot_face)        
+        self.draw_labyr(self.labyr.labyr)
```

Comment:
Drawing a new labyrinth is not necessary. Use the previously generated in th init instead.